contracts/Prover.sol

```diff
@@ -90,29 +81,33 @@ contract Prover is SimpleProver, AbstractProver {
     // But optimism maintains a service that posts L1 block data on L2.
     IL1Block public l1BlockhashOracle;
 
-    enum ProvingMechanism {
-        Self, // Used for Ethereum and Sepolia (any chain that settles to itself)
-        Bedrock,
-        Cannon,
-        Arbitrum,
-        HyperProver
+    struct ChainConfigurationKey {
```

Comment:
wondering if there is a way to thin out the number of structs like this, its quite confusing to follow this pattern for construction. Can we put the chainID and provingMechanism in the chainConfiguration? I'd be down to have a working sesssion abotu these and figure out if we cant make them a little cleaner.